osf/models/preprint_service.py

```diff
@@ -259,3 +281,307 @@ def _send_preprint_confirmation(self, auth):
             node=self.node,
             preprint=self
         )
+
```

Comment:
Instead of copying over all of these permissions methods, how about something like a `PermissionsMixin` to avoid duplicated code? That way, if we need to make changes it can just happen in one place